eureka/S4_generate_lightcurves/s4_genLC.py

```diff
@@ -112,29 +116,30 @@ def lcJWST(eventlabel, ecf_path='./', s3_meta=None):
             meta.copy_ecf()
 
             log.writelog("Loading S3 save file", mute=(not meta.verbose))
-            table = astropytable.readtable(meta.tab_filename)
+            #table = astropytable.readtable(meta.tab_filename)
+            spec = xrio.readXR(meta.filename_S3_SpecData)
 
             # Reverse the reshaping which has been done when saving the astropy table
-            optspec = np.reshape(table['optspec'].data, (-1, meta.subnx))
-            opterr = np.reshape(table['opterr'].data, (-1, meta.subnx))
-            wave_1d = table['wave_1d'].data[0:meta.subnx]
-            meta.time = table['time'].data[::meta.subnx]
+            # optspec = np.reshape(table['optspec'].data, (-1, meta.subnx))
+            # opterr = np.reshape(table['opterr'].data, (-1, meta.subnx))
+            # wave_1d = table['wave_1d'].data[0:meta.subnx]
+            # meta.time = table['time'].data[::meta.subnx]
 
             if meta.wave_min is None:
-                meta.wave_min = np.min(wave_1d)
+                meta.wave_min = np.min(spec.wave_1d.values)
                 log.writelog(f'No value was provided for meta.wave_min, so defaulting to {meta.wave_min}.', mute=(not meta.verbose))
-            elif meta.wave_min<np.min(wave_1d):
-                log.writelog(f'WARNING: The selected meta.wave_min ({meta.wave_min}) is smaller than the shortest wavelength ({np.min(wave_1d)})')
+            elif meta.wave_min<np.min(spec.wave_1d.values):
+                log.writelog(f'WARNING: The selected meta.wave_min ({meta.wave_min}) is smaller than the shortest wavelength ({np.min(spec.wave_1d.values)})')
             if meta.wave_max is None:
-                meta.wave_max = np.max(wave_1d)
+                meta.wave_max = np.max(spec.wave_1d.values)
                 log.writelog(f'No value was provided for meta.wave_max, so defaulting to {meta.wave_max}.', mute=(not meta.verbose))
-            elif meta.wave_max>np.max(wave_1d):
-                log.writelog(f'WARNING: The selected meta.wave_max ({meta.wave_max}) is larger than the longest wavelength ({np.max(wave_1d)})')
+            elif meta.wave_max>np.max(spec.wave_1d.values):
+                log.writelog(f'WARNING: The selected meta.wave_max ({meta.wave_max}) is larger than the longest wavelength ({np.max(spec.wave_1d.values)})')
 
             #Replace NaNs with zero
-            optspec[np.where(np.isnan(optspec))] = 0
-            opterr[np.where(np.isnan(opterr))] = 0
-            meta.n_int, meta.subnx   = optspec.shape
+            spec.optspec[np.where(np.isnan(spec.optspec))] = 0
+            spec.opterr[np.where(np.isnan(spec.opterr))] = 0
```

Comment:
Why do we do this?

Reply:
I think this is from before we implemented masked arrays.  Not sure it serves a purpose anymore.

Reply:
If anything, this might be causing some of our issues with masked arrays